R/trimData.R

```diff
@@ -28,23 +28,39 @@ trimData <- function(data, settings, chart="edish"){
 
   # Add items in vectors to list individually
   dataVectorKeys <- allKeys %>% filter(.data$setting_type =="vector") %>% pull(.data$text_key) %>% textKeysToList()
+  
+  vectorVars <- list()
+  
   for(key in dataVectorKeys){
     current<-getSettingValue(key, settings=settings)
+
     if (length(current) > 0 ) {
-      for (i in 1:length(current)){
-        newKey <- key
-        newKey[[1+length(newKey)]]<-i
-        sub <- current[[i]]
-        if(typeof(sub)=="list"){
-          newKey[[1+length(newKey)]]<-"value_col"
+      # handle vectors separately
+      if (typeof(current) == "character") {
+        for (i in 1:length(current)){
+          vectorVars <- append(vectorVars,current[i])
+        }
+      } else {
+        for (i in 1:length(current)){
+          newKey <- key
+          newKey[[1+length(newKey)]]<-i
+          sub <- current[[i]]
+          if(typeof(sub)=="list"){
+            newKey[[1+length(newKey)]]<-"value_col"
+          }
+          dataKeys[[1+length(dataKeys)]]<-newKey
         }
-        dataKeys[[1+length(dataKeys)]]<-newKey
       }
     }
   }
 
-  settings_values <- map(dataKeys, function(x) {return(getSettingValue(x, settings))})
 
+  
+  settings_values <- map(dataKeys, function(x) {return(getSettingValue(x, settings))})
```

Comment:
for the filters etc. that are true vectors getSettingValue returns all the values for the keys so im not sure if going through this set up would work